test/units/modules/network/checkpoint/test_cp_network.py

```diff
@@ -26,22 +26,22 @@
 from ansible.modules.network.checkpoint import cp_network
 
 
-OBJECT = {'name': 'test_network', 'nat_settings': {'auto-rule': True,
-                                                   'hide-behind': 'ip-address',
-                                                   'ip-address': '192.168.1.111'},
+OBJECT = {'name': 'test_network', 'nat_settings': {'auto_rule': True,
+                                                   'hide_behind': 'ip-address',
+                                                   'ip_address': '192.168.1.111'},
```

Comment:
```suggestion
OBJECT = {'name': 'test_network', 'nat_settings': [{'auto_rule': True,
                                                   'hide_behind': 'ip-address',
                                                   'ip_address': '192.168.1.111'}],
```

Reply:
change the dict to list of dict for both Create and Update option and it should pass then